src/modalities/api.py

```diff
@@ -0,0 +1,131 @@
+#!/usr/bin/env python
+
+import os
+from pathlib import Path
+
+from pydantic import FilePath
+
+import modalities.inference.inference as inference
+from modalities.checkpointing.checkpoint_conversion import CheckpointConversion
+from modalities.config.component_factory import ComponentFactory
+from modalities.config.config import load_app_config_dict
+from modalities.config.instantiation_models import PackedDatasetComponentsInstantiationModel
+from modalities.dataloader.create_index import IndexGenerator
+from modalities.dataloader.create_packed_data import EmbeddedStreamData, PackedDataGenerator, join_embedded_stream_data
+from modalities.dataloader.large_file_lines_reader import LargeFileLinesReader
+from modalities.models.huggingface_adapters.hf_adapter import HFModelAdapter
+from modalities.registry.components import COMPONENTS
+from modalities.registry.registry import Registry
+
+
+def create_raw_data_index(src_path: Path, index_path: Path):
+    """Creates the index file for the content of a large jsonl-file. The index file
+    contains the byte-offsets and lengths of each line in the jsonl-file.
+    Background is the ability to further process the respective file without loading it,
+    while splitting its content line-based. This step is necessary in advance of further processing like tokenization.
+    It is only necessary once for a jsonl-file and allows therefore different tokenizations without re-indexing.
+
+    Args:
+        src_path (Path): The path to the jsonl-file.
+        index_path (Path): The path to the index file, that will be created.
+
+    Raises:
+        ValueError: If the index file already exists.
+    """
+    index_path = LargeFileLinesReader.default_index_path(src_path, index_path)
+    os.makedirs(index_path.parent, exist_ok=True)
+    if index_path.exists():
+        raise ValueError("index already exists. delete it or specify different output folder.")
+
+    print(f"reading raw data from {src_path}")
```

Comment:
The code contains multiple print-statements. Shouldn't we use our logging-functionality or the logging-module instead?

Reply:
I think it is fine for now. It is just for the index generation and printed only once.